components/salsa-2022-macros/src/input.rs

```diff
@@ -73,8 +87,16 @@ impl InputStruct {
     }
 
     fn validate_input(&self) -> syn::Result<()> {
+        // check for dissalowed fields
         self.disallow_id_fields("input")?;
 
+        // check if an input struct labeled singleton truly has one field
+        if self.0.is_isingleton() && self.0.num_fields() != 1 {
```

Comment:
Why can't we support multiple fields here?

Reply:
I'll work on the tests. I assumed wrongly that Singleton structs should have only one field. I see that it's wrong now. Will remove it.
Thanks!

Reply:
I agree the limitation is not needed. Sounds good.

Reply:
Thanks for the comment and recommendation